src/main/java/org/verdictdb/coordinator/VerdictSingleResultFromListData.java

```diff
@@ -2,34 +2,47 @@
 
 import com.google.common.base.Optional;
 import com.rits.cloning.Cloner;
+import org.verdictdb.commons.AttributeValueRetrievalHelper;
 import org.verdictdb.connection.DataTypeConverter;
 import org.verdictdb.connection.DbmsQueryResult;
 import org.verdictdb.connection.DbmsQueryResultMetaData;
 
 import java.io.*;
 import java.sql.Types;
+import java.util.ArrayList;
 import java.util.List;
 
-public class VerdictSingleResultFromListData extends VerdictSingleResult {
+public class VerdictSingleResultFromListData
+    extends AttributeValueRetrievalHelper implements VerdictSingleResult {
 
   private Optional<List<Object>> result;
 
+  List<String> fieldsName = new ArrayList<>();
+
   // used to support wasnull()
   private Object lastValueRead;
 
   int cursor = -1;
 
-  public VerdictSingleResultFromListData(List<Object> result) {
+  public VerdictSingleResultFromListData(ExecutionContext.QueryType type, List<Object> result) {
```

Comment:
Please change the first argument type to `List<String>` and the second type to `List<List<Object>>`.

You can add another factory method that gets `List<Object>` in the second argument, assuming that the table has only one column.